internal/venice-client-common/src/main/java/com/linkedin/venice/stats/VeniceOpenTelemetryMetricsRepository.java

```diff
@@ -282,20 +286,54 @@ public LongCounter createCounter(MetricEntity metricEntity) {
     });
   }
 
-  public DoubleGauge createGuage(MetricEntity metricEntity) {
+  public LongGauge createGuage(MetricEntity metricEntity) {
     if (!emitOpenTelemetryMetrics()) {
       return null;
     }
     return gaugeMap.computeIfAbsent(metricEntity.getMetricName(), key -> {
       String fullMetricName = getFullMetricName(metricEntity);
-      DoubleGaugeBuilder builder = meter.gaugeBuilder(fullMetricName)
+      LongGaugeBuilder builder = meter.gaugeBuilder(fullMetricName)
           .setUnit(metricEntity.getUnit().name())
-          .setDescription(getMetricDescription(metricEntity, metricsConfig));
+          .setDescription(getMetricDescription(metricEntity, metricsConfig))
+          .ofLongs();
       return builder.build();
     });
   }
 
-  public Object createInstrument(MetricEntity metricEntity) {
+  /**
```

Comment:
While this looks interesting, it looks like for now we're not using this for anything outside of tests, and anecdotally I think that means it'll be unlikely that developers adding features will be likely to use it.

Is there a metric that exists today that could stand to use this?  Or... is there something coming in a follow up PR that will use it?

Reply:
(I'm referring to the async gauge, I think I mislicked the line number but I guess I'm too lazy to fix it)